bin/registerServerTopology.sh

```diff
@@ -0,0 +1,373 @@
+    --insecure <optional>       allow self-signed certificate
+    --template <optional>       provide the path of a register topology configuration file to automate the script
+              - API_USERNAME             <mandatory> string
+              - API_TARGET_PASSWORD      <mandatory> string
+              - CURRENT_NODE_TYPE        <mandatory> string
+              - TARGET_NODE_ADDRESS      <mandatory> string (PARENT NODE ADDRESS)
+              - CURRENT_NODE_NAME        <mandatory> string (CURRENT NODE NAME)
+              - CURRENT_NODE_ADDRESS     <mandatory> string (CURRENT NODE IP OR FQDN)
+              - ROOT_CENTREON_FOLDER     <optional> string (CENTRAL ROOT CENTREON FOLDER)
+              - INSECURE                 <optional> boolean
+
+EOF
+}
+#========= end of function usage()
+
+
+#========= begin of get_api_password()
+function get_api_password() {
+  stty -echo
+  echo "${TARGET_NODE_ADDRESS} : Please enter your password"
+  read -r API_TARGET_PASSWORD
+  stty echo
+}
+#========= end of get_api_password()
+
+#========= begin of get_current_node_ip()
+function get_current_node_ip() {
+  CURRENT_NODE_ADDRESS=$(hostname -I)
+
+  ips=($CURRENT_NODE_ADDRESS)
+  count_available_ips=${#ips[@]}
+
+  if [[ $count_available_ips -gt 1 ]];
+  then
+    echo "Which IP do you want to use as CURRENT NODE IP ?"
+    for i in "${!ips[@]}";
+    do
+      printf "%s) %s\n" "$i" "${ips[$i]}"
+    done
+
+    read -r choice
+
+    if [[ $choice -ge 0 && $choice -le $count_available_ips-1 ]];
+    then
+      CURRENT_NODE_ADDRESS=${ips[$choice]}
+    else
+      get_current_node_ip
+    fi
+  fi
+}
+#========= end of get_current_node_ip()
+
+
+#========= begin of prepare_register_payload()
+# Format all the information in JSON Format and display a reminder of all sent information
+function prepare_register_payload() {
+  PAYLOAD='{"name":"'"${CURRENT_NODE_NAME}"'","hostname":"'"${HOSTNAME}"'","type":"'"${CURRENT_NODE_TYPE}"'","address":"'"${CURRENT_NODE_ADDRESS}"'","parent_address":"'"${PARSED_URL[HOST]}"'"}'
+
+  cat << EOD
+
+  Summary of the information that will be send:
+
+  Api Connection:
+  username: ${API_USERNAME}
+  password: ******
+  target server: ${PARSED_URL[HOST]}
+
+  Pending Registration Server:
+  name: ${CURRENT_NODE_NAME}
+  hostname: ${HOSTNAME}
+  type: ${CURRENT_NODE_TYPE}
+  address: ${CURRENT_NODE_ADDRESS}
+
+EOD
+
+  echo 'Do you want to register this server with those information? (y/n) '
+  read -r IS_VALID
+
+  if [[ $IS_VALID != 'y' ]];
+  then
+    log "INFO" "Registration aborted"
+    exit 0
+  fi
+}
+#========= end of prepare_register_payload()
+
+
+#========= begin of register_server()
+# Send the request to register the server
+function register_server() {
+  IFS=$'\n' API_RESPONSE=($(curl -s -X POST ${INSECURE:+--insecure} -i -H "Content-Type: application/json" -H "X-AUTH-TOKEN: ${API_TOKEN}" \
```

Comment:
$IFS is a special shell variable, you should rename it :
https://bash.cyberciti.biz/guide/$IFS

Reply:
as discussed, overriding the IFS is required -> so you should explain the intention by adding a comment

Reply:
without IFS:

2020-12-16 09:54:57+01:00 - ERROR - 409

with:

2020-12-16 09:43:26+01:00 - ERROR - {"message":"A platform using the name : 'xxxx' or address : 'xxx.xx.x.x' already exists"}

I'll comment this portion of code to explain it